starter-core/src/main/java/io/micronaut/starter/feature/Features.java

```diff
@@ -99,11 +99,7 @@ public JdkVersion javaVersion() {
     }
 
     public String getTargetJdk() {
-        if (language().isKotlin() || testFramework().isKoTest()) {
-            return VersionInfo.toJdkVersion(Math.min(javaVersion.majorVersion(), 14));
-        } else {
-            return VersionInfo.toJdkVersion(javaVersion.majorVersion());
-        }
+        return VersionInfo.toJdkVersion(javaVersion.majorVersion());
```

Comment:
I've tried a generated Kotlin app with Java 16 and it doesn't work. It fails with

```
> Task :kaptGenerateStubsKotlin
e: Unknown JVM target version: 16
Supported versions: 1.6, 1.8, 9, 10, 11, 12, 13, 14, 15

> Task :kaptGenerateStubsKotlin FAILED

FAILURE: Build failed with an exception.
```